docs/internals/details.rst

```diff
@@ -0,0 +1,36 @@
+Implementation Details
+======================
+
+
+.. _log_density_propto:
+
+``log_density`` with ``propto=true``
+------------------------------------
+
+The log density function provided by a Stan model has
```

Comment:
I would phrase this as having the ability to drop constants.   Then I'd give simple recommendations:

1.  If you're running MCMC that needs gradients and only density up to proportion, then use `propto = true`.  Setting `propto=true` will be at least as fast.
2.  To evaluate the log density on `double` values to match, use `propto=true`.  Setting `propto=true` may be slower or faster, depending on the cost of calculating normalizing constants (`propto=false`) and the cost of autodiff (required to get the right answer if `propto=true`).

I don't think we need to say much more than that.

Why the double back ticks?  

I couldn't understand what lines 34/35 were doing.




I'd just give simple recommendations:

* doing gradient-based calculations with autodiff:  use propto because it's faster
* doing log density evals without autodiff:  depending

Reply:
The double back ticks are how ReStructuredText (that sphinx uses) wants code formatted. They're equivalent to single backticks in Markdown. Lines 34/35 are also a RST detail to get a link that is also `code formatted`. The result is that "|reduce_sum|" above gets rendered as [`reduce_sum`](https://mc-stan.org/docs/stan-users-guide/reduce-sum.html)

I agree phrasing it in terms of a suggestion for each case is clearer. I left the explanation in, but under a sub-heading for the curious.